drivers/sensor/examplesensor/CMakeLists.txt

```diff
@@ -0,0 +1,4 @@
+if(CONFIG_EXAMPLESENSOR)
```

Comment:
This is a common problem with upstream drivers, but the `if(CONFIG_EXAMPLESENSOR)` is redundant here, bceause the whole subdirectory is excluded if the config option is n, in the parent directory CMake list file:

```
 add_subdirectory_ifdef(CONFIG_EXAMPLESENSOR examplesensor)
```

It would be nice not to continue encouraging people to copy paste that